src/main/java/frc/robot/subsystems/Lift.java

```diff
@@ -49,17 +54,20 @@ public Lift(RobotMap.LiftMap map) {
         liftEncoder = map.getLiftEncoder();
         elevatorBrake = map.liftBrake();
         upperLimitSwitch = map.upperLiftLimit();
+        lowerLimitSwitch = map.lowerLiftLimit();
     }
 
     // sets the ratchet to either be activated or deactivated depending on liftSpeed
     public void liftSpeed(double speed) {
-        if (speed > 0) {
-            elevatorBrake.set(false);
-            if (upperLimitSwitch.getAsBoolean()) {
+        if (Math.abs(speed) <= .1) {
+            speed = 0.0;
+        }
+        if (speed < 0) {
+            if (upperLimitSwitch.getAsBoolean() || lowerLimitSwitch.getAsBoolean() || !elevatorBrake.get()) {
```

Comment:
Only one limit switch should be here. The `lowerLimitSwitch` should be checked in the below `if (speed > 0)`.